### PR TITLE
fix(pulsar): handle `Redirect` in `LookupTopicResponse`

### PR DESCRIPTION
**Pull request by thalesmg**
See https://github.com/emqx/pulsar-client-erl/pull/68

Fixes https://emqx.atlassian.net/browse/EMQX-13572

Release version: e5.9

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_bridge_pulsar/test/emqx_bridge_pulsar_v2_SUITE.erl [hunk 1]
```diff
@@ -275,7 +275,7 @@ start_consumer(TestCase, Config) ->
         cacertfile => filename:join([CertsPath, "cacert.pem"])
     },
     Opts = #{enable_ssl => UseTLS, ssl_opts => emqx_tls_lib:to_client_opts(SSLOpts)},
-    {ok, _ClientPid} = pulsar:ensure_supervised_client(ConsumerClientId, [URL], Opts),
+    {ok, _} = pulsar:ensure_supervised_client(ConsumerClientId, [URL], Opts),
```
Comment (savonarola) [inline apps/emqx_bridge_pulsar/test/emqx_bridge_pulsar_v2_SUITE.erl:278]: It is not pid anymore?

Comment (thalesmg) [reply apps/emqx_bridge_pulsar/test/emqx_bridge_pulsar_v2_SUITE.erl:278]: It is, but I went back and forth in changing the driver while testing with EMQX, ended up changing the variable binding name.

**Comment by thalesmg** [comment]
The CI issue is unrelated with the changes herein.

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-6686)

**Pull request merged**